mbed_greentea/mbed_target_info.py

```diff
@@ -142,17 +142,13 @@ def parse_yotta_json_for_build_name(yotta_json_content):
     @param yotta_json_content Content of .yotta_json file
     @return String with set yotta target name, None if no target found
     """
-    result = None
-    if 'build' in yotta_json_content:
-        if 'target' in yotta_json_content['build']:
-            # yotta_json_content['build']['target'] should be something like this: "frdm-k64f-gcc,*"
-            # We will split this string using ','. Because split will always return an array
-            # with at least 1 element we can select first element to be target name
-            try:
-                result = yotta_json_content['build']['target'].split(',')[0]
-            except IndexError:
-                result = None
-    return result
+    if 'build' not in yotta_json_content:
+        return None
+
+    if 'target' not in yotta_json_content['build']:
+        return None
+
+    return yotta_json_content['build']['target'].split(',')[0]
```

Comment:
The try/catch block was removed here, was this intentional?

Reply:
As far as I could tell, it can't throw, the keys are checked in the if statements before and split always seems to return a list with a value, even if it's an empty string

Reply:
@scartmell-arm It's more pythonic (and slightly faster) to just go for the indexing and catch the exception.

Reply:
@theotherjimmy Okay, in that case I'll change it to what you suggest somewhere else in the comments